src/main/java/duke/Command.java

```diff
@@ -0,0 +1,203 @@
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.util.DukeException;
+
+
+/**
+ * Command class that handles the logic flow of commands input.
+ */
+
+public class Command {
+
+    private final String[] input;
+    private boolean shouldExit = false;
+
+    /**
+     * Constructor that creates a Command object.
+     * @param input command string that has been split
+     */
+
+    public Command(String[] input) {
+        this.input = input;
+    }
+
+    /**
+     * Evaluates the syntax of the input command and executes it, throwing
+     * a DukeException if the syntax is incorrect.
+     * @param tasks the TaskList object associated with the current Duke object
+     * @param ui the Ui object associated with the current Duke object
+     * @param storage the Storage object associated with the current Duke object
+     * @throws DukeException thrown if any errors are detected with the input
+     */
+
+    public void execute(TaskList tasks, Ui ui, Storage storage)
+            throws DukeException {
+        switch(input[0]) {
+        case "bye":
+            ui.printBye();
+            this.shouldExit = true;
+            break;
+        case "list": {
+            ui.printList(tasks);
+            break;
+        }
+        case "done": {
+            int index;
+            if (input.length == 1) {
+                throw new DukeException("    Please select a task to mark as completed!");
+            }
+            try {
+                index = Integer.parseInt(input[1]);
+            } catch (NumberFormatException e) {
+                throw new DukeException("    Please choose an integer value!");
+            }
+            if (index <= 0) {
+                throw new DukeException("    Please choose an integer greater than 0!");
+            } else if (index > tasks.size()) {
+                throw new DukeException("    Your task list is not that long yet!");
+            }
+            tasks.completeTask(index);
+            ui.printDone(tasks.getTask(index));
+            storage.save(tasks);
+            break;
+        }
+        case "delete": {
+            int index;
+            if (input.length == 1) {
+                throw new DukeException("    Please select a task to mark as completed!");
+            }
+            try {
+                index = Integer.parseInt(input[1]);
+            } catch (NumberFormatException e) {
+                throw new DukeException("    Please choose an integer value!");
+            }
+            if (index <= 0) {
+                throw new DukeException("    Please choose an integer greater than 0!");
+            } else if (index > tasks.size()) {
+                throw new DukeException("    Your task list is not that long yet!");
+            }
+            Task task = tasks.getTask(index);
+            tasks.deleteTask(index);
+            ui.printDelete(tasks, task);
+            storage.save(tasks);
+            break;
+        }
+        case "find": {
```

Comment:
Based on coding standards, I think no need to put curly brackets for case block.

Reply:
Curly brackets were necessary as multiple object definitions of the same name would not be permitted otherwise (for instance index or task).